src/poprox_recommender/recommenders/__init__.py

```diff
@@ -33,23 +34,30 @@ def select_articles(
     pipeline_params: dict[str, Any] | None = None,
 ) -> PipelineState:
     """
-    Select articles with default recommender configuration.  It returns a
-    pipeline state whose ``default`` is the final list of recommendations.
+    Select articles with default recommender configuration. It returns a
+    pipeline state whose "image-selector" key contains the final list of recommendations
+    with selected images.
     """
     name = None
     if pipeline_params and "pipeline" in pipeline_params:
         name = pipeline_params["pipeline"]
-
     if name is None:
         name = default_pipeline()
 
     pipeline = get_pipeline(name)
 
     recs = pipeline.node("recommender")
+    image_selector = pipeline.node("image-selector")
     topk = pipeline.node("ranker", missing="none")
+
     if topk is None:
-        wanted = (recs,)
+        wanted = (image_selector, recs)
```

Comment:
By making the output of the image selector the output of the whole pipeline, we'll get the articles with personalized images in the `recs`. This file shouldn't need changes to make your recommender pipeline work.